azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -432,10 +428,16 @@ public void submitFlow(final int execId) throws ExecutorManagerException {
       if (this.executorService.isShutdown()) {
         errorMsg.append("The executor is being shut down.");
       }
-      throw new ExecutorManagerException(errorMsg.toString(), re);
+      throw new ExecutorResponseErrorException(errorMsg.toString(), re,
+          ResponseErrorType.SERVICE_UNAVAILABLE);
     }
   }
 
+  private static ExecutorResponseErrorException createAlreadyRunningException(final int execId) {
+    return new ExecutorResponseErrorException("Execution " + execId + " is already running.",
+        ResponseErrorType.BAD_REQUEST);
```

Comment:
BAD_REQUEST is actually a more demanding instruction than "_should_ stop dispatching". In this case client "_must_ stop dispatching". This is used to prevent dispatching the same execution on multiple executors.

Now that I think of it, maybe there should be a separate code for that case, though. For example: `ResponseErrorType.ALREADY_RUNNING`. That way the client could handle it gracefully: for example if it previously tried to dispatch but got a network timeout, then it could deem it as a successful dispatch if executor says that the execution had been dispatched on the previous attempt.